cpp/cmake/thirdparty/get_faiss.cmake

```diff
@@ -30,36 +30,57 @@ function(find_and_configure_faiss)
             set(CPM_DOWNLOAD_faiss ON)
         endif()
 
+        include(cmake/modules/FindAVX.cmake)
+
+        # Link against AVX CPU lib if it exists
+        set(RAFT_FAISS_GLOBAL_TARGETS faiss::faiss)
```

Comment:
You don't need these conditionals around GLOBAL_TARGETS just provide them all and rapids-cmake will first check that they exist before doing any operations on them